src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -52,7 +52,7 @@ public static byte[] DecodeHexString(this string s)
             bool byteInProgress = false;
             int index = 0;
 
-            for (int i = 0; i < s.Length; i++)
+            for (int i = s.Length != 0 && s[0] == '\u200E' ? 1 : 0; i < s.Length; i++)
```

Comment:
If the string started with `\u200E` and had even length `cbHex` will be one larger than it should be, impacting the `TestBySubjectKeyIdentifier_Compat` test and friends.


Since this project is only building against targets that have Span, I'd fix this as

```C#
         public static byte[] DecodeHexString(this string str)
         {
             int whitespaceCount = 0;
             ReadOnlySpan<char> s = str;

             if (s.Length > 0 && s[0] == '\u200E')
             {
                 s = s.Slice(1);
             }

             for (...)
```

(Rename the input parameter, add a ReadOnlySpan<char>, and conditionally slice it)